jax/_src/util.py

```diff
@@ -199,7 +199,7 @@ def wrapper(*args, **kwargs):
       if config.jax_check_tracer_leaks:
         return f(*args, **kwargs)
       else:
-        return cached(config._trace_context(), *args, **kwargs)
+        return cached(config.get_thread_local_trace_state(), *args, **kwargs)
```

Comment:
I don't think this part is correct. In general you need to merge the thread-local state with the global state to get a cache key, I think?

Reply:
Ah good, this is exactly the kind of thing I wanted to check!

LMK if you have a pointer to where this is already done somewhere (even if in C++) so I can check against it...

Reply:
We think this is actually correct: the other global parts of the state are already introduced in this function by it reading the relevant attributes from the `config` object. But the function name was misleading since it included "thread_local". I'll change the name!

Separately, I may look at making this faster...